python/common/org/python/types/Slice.java

```diff
@@ -35,8 +35,13 @@ public Slice(org.python.Object start, org.python.Object stop, org.python.Object
         }
 
         if (step instanceof org.python.types.Int) {
-            this.__dict__.put("step", step);
             this.step = (org.python.types.Int) step;
+            if (this.step.value == 0 ){
+                throw new org.python.exceptions.ValueError("slice step cannot be zero");
+            }
+            else{
```

Comment:
Another spacing issue here; the else should be on the same line as the previous bracket, and there should be a space before and after else - i.e., `} else {`